CMakeLists.txt

```diff
@@ -1,22 +1,48 @@
 cmake_minimum_required(VERSION 3.20)
 project(AxleNyxus)
 
-option(USEGPU "Use GPU" OFF) 
+find_package(CUDA)
+if(CUDA_FOUND)
+    try_run(RUN_RESULT_VAR COMPILE_RESULT_VAR
+        ${CMAKE_BINARY_DIR} 
+        ${CMAKE_CURRENT_SOURCE_DIR}/src/nyx/gpu/find_cuda_gpu.c
+        CMAKE_FLAGS 
+            -DINCLUDE_DIRECTORIES:STRING=${CUDA_TOOLKIT_INCLUDE}
+            -DLINK_LIBRARIES:STRING=${CUDA_CUDART_LIBRARY}
+        COMPILE_OUTPUT_VARIABLE COMPILE_OUTPUT_VAR
+        RUN_OUTPUT_VARIABLE RUN_OUTPUT_VAR)
+	message("---------${CUDA_TOOLKIT_INCLUDE}--------")
+    message("${RUN_OUTPUT_VAR}") # Display number of GPUs found
+    # COMPILE_RESULT_VAR is TRUE when compile succeeds
+    # RUN_RESULT_VAR is zero when a GPU is found
+    if(COMPILE_RESULT_VAR AND NOT RUN_RESULT_VAR)
+		option(USEGPU "Use GPU" ON) 
+    else()
+		option(USEGPU "Use GPU" OFF) 
+    endif()
+else()
+	option(USEGPU "Use GPU" OFF) 
+endif(CUDA_FOUND)
+
 if(USEGPU)
     set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
     set(CMAKE_CUDA_HOST_LINK_LAUNCHER ${CMAKE_CXX_COMPILER})
     enable_language("CUDA")
     add_definitions(-DUSE_GPU)
 
-    #---Additionally, the architecture can be set manually eg sm_37 for K80--- 
-    #	set(CMAKE_CUDA_ARCHITECTURES 37 CACHE STRING "CUDA architectures" FORCE)
+    include(FindCUDA/select_compute_arch)
+	CUDA_DETECT_INSTALLED_GPUS(INSTALLED_GPU_CCS_1)
+	string(STRIP "${INSTALLED_GPU_CCS_1}" INSTALLED_GPU_CCS_2)
+	string(REPLACE " " ";" INSTALLED_GPU_CCS_3 "${INSTALLED_GPU_CCS_2}")
+	string(REPLACE "." "" CUDA_ARCH_LIST "${INSTALLED_GPU_CCS_3}")
+	SET(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH_LIST})
+	set_property(GLOBAL PROPERTY CUDA_ARCHITECTURES "${CUDA_ARCH_LIST}")
+	message("Found GPU with compute capability ${CUDA_ARCH_LIST}.")
 endif()
 
 #==== Compiler Options
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
-set(CMAKE_CUDA_STANDARD 17)
-set(CMAKE_CUDA_STANDARD_REQUIRED True)
```

Comment:
Are we dropping these lines intentionally?

Reply:
Those lines were duplicates. The same lines appear on lines 16-17 and 18-19 in the current CMake so I removed the duplicate. Let me know if you want them added back.